mondrian/src/main/java/mondrian/xmla/XmlaHandler.java

```diff
@@ -842,6 +954,8 @@ private QueryResult executeQuery( XmlaRequest request ) throws XmlaException {
       } catch ( Exception ex ) {
         throw new XmlaException( SERVER_FAULT_FC, HSB_EXECUTE_QUERY_CODE, HSB_EXECUTE_QUERY_FAULT_FS, ex );
       }
+    } catch ( OlapException e ) {
+      throw new RuntimeException( e );
```

Comment:
Instead directly throwing `RuntimeException`, maybe throw `MondrianException` (or other that extends RE) to allow more flexibility?